### ISSUE TITLE
SwiftLint tool should fail builds if accessibility tags are not present for UIControls in iOS

### ISSUE THREAD

**Issue by KarthikaThevan** [realm/SwiftLint#3508]
Hi Team,

We are having ebook reading application developed in IOS platform. We want to use swiftLint tool to check whether accessibility enabled for all UIControls which are in storyboard as well as programmatically initialised controls.

Our requirement: We are supporting voice over for visually disabled people. Accessibility should be enabled for all UI controls (eg : UIButton, UILabel… etc) in application to support voice over.

Use case : When we make a build, if the controls doesn’t have accessibility enabled true property, then the build should fail in Xcode. SwiftLint tool should fail builds if accessibility tags are not present for UIControls in the application.

Kindly guide us how to get all UIElements in the application and how to check the property values of the UIElement instance. 
For example, get all the UILabels and check whether the UILabels are set as IsAccessibilityElement is True. Likewise, we need to get all UIElements and check the IsAccessibilityElement property value for each UIElement instance. 

Could you please suggest on the above so that, we can try our best to implement it in SwiftLint.

**Comment by KarthikaThevan** [comment]
Hi Team, 

Could you help on this?

**Comment by jpsim** [comment]
This would be a tricky rule to build. Maybe impossible. To be accurate, it would probably need to be built as an analyzer rule, and somehow determine that there's a `UIControl` in the class hierarchy for a given reference, and that somewhere there's a call to set the accessibility tags.

I'll close this because I don't think it's reasonably possible to build a rule that checks this well today.
